src/libsyntax/ext/tt/macro_rules.rs

```diff
@@ -181,7 +181,7 @@ fn generic_extension<'cx>(cx: &'cx mut ExtCtxt,
         for lhs in lhses { // try each arm's matchers
             let lhs_tt = match *lhs {
                 quoted::TokenTree::Delimited(_, ref delim) => &delim.tts[..],
-                _ => cx.span_bug(sp, "malformed macro lhs")
+                _ => continue,
```

Comment:
This is in the recovery path, we shouldn't be hitting this ever, as it would be caught above, but if we were we don't really care.